android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -358,6 +360,28 @@ public void getSystemAvailableFeatures(Promise p) {
     p.resolve(promiseArray);
   }
 
+  @ReactMethod
+  public void hasLocationServicesEnabled(Promise p) {
+      boolean locationEnabled = false;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT){
+          try {
+              int locationMode = Settings.Secure.getInt(this.reactContext.getContentResolver(), Settings.Secure.LOCATION_MODE);
```

Comment:
Can you use the mode with a default value since we default to Settings.Secure.LOCATION_MODE_OFF anyway? That way we can avoid the possibility of exception, get rid of the try/catch (and what made me think about it) not have to decide whether to printStackTrace or p.reject() etc

Reply:
also - it's peppered in the code a bit, but in javascript 'this.' references are redundant